overlord/snapstate/handlers.go

```diff
@@ -1155,8 +1155,8 @@ func maybeRestart(t *state.Task, info *snap.Info) {
 	}
 
 	typ := info.GetType()
-	bp, applicable := boot.Lookup(info, typ, model, release.OnClassic)
-	if applicable {
+	bp := boot.Lookup(info, typ, model, release.OnClassic)
+	if !bp.IsTrivial() {
```

Comment:
this seems the only non test place that uses that,

also can't it be written as everything else? :

if !boot.Participant(info, typ, model, release.OnClassic).ChangeRequiresReboot() { ...

Reply:
right now yes, because `daemonRestartReason` will always return `""` in the cases where `boot.Participant()` returns something non-trivial whose `ChangeRequiresReboot()` returns `false`. But that feels more coincidental than anything else, so I'm not sure we want to code it that way... but open to being convinced :-)

Reply:
actually: `daemonRestartReason` will return `""` for _any_ non-trivial `Participant`, because for a non-empty reason either it's classic or it's snapd.
Maybe a comment is enough.

Reply:
added a comment